### DISCUSSION TITLE
Package and host requirements?

### DISCUSSION THREAD

**Discussion by if-nick** [spatie/ssh#81] [Q&A]
Could you share host or server requirements for using this library? Is a php ssh extension required? 

On our host env we have ssh: 
```
$ whereis ssh
ssh: /usr/bin/ssh /etc/ssh /usr/share/man/man1/ssh.1.gz
```

But on a web page on the host we do not. For example: 
```
<?php 
/***
 * https://..../test.php
 */
echo exec("which ssh");
// No output.
```

Similar results after installing the library. The Process result is statuscode 127, command not found: 
```
<?php

namespace Drupal\my_test\Controller;

use GuzzleHttp\ClientInterface;
use Symfony\Component\Process\Process;
use Spatie\Ssh\Ssh;

/**
 * Fetch a token.
 */
class GetSSHToken {

  /**
   * Fetch the SSH result.
   */
  public static function test() {

    dump([
      exec("ssh"),
      exec("which ssh"),
    ]);

    $process = Ssh::create($sshUser, $sshHost)
      ->usePort($sshPort)
      ->usePrivateKey($sshKeyPath)
      ->disableStrictHostKeyChecking()
      ->execute('token');

    dump([
      $process,
      $process->isSuccessful(),
      $process->getOutput(),
    ]);

  }
}
```

**Comment by asciitoo** [comment]
This package doesn't require any SSH extension, you should check if the SSH command is available. The package uses the `symfony/process` so you could execute the commands available.

I'm not an expert in Linux, actually, I'm a total beginner, but my recommendation is to check your hosting if the web user/group has access to SSH command.
